quill-core/src/main/scala/io/getquill/context/ActionMacro.scala

```diff
@@ -135,17 +137,16 @@ class ActionMacro(val c: MacroContext)
 
   private def returningColumn =
     q"""
-      expanded.ast match {
-        case io.getquill.ast.Returning(_, _, io.getquill.ast.Property(_, property)) =>
-          expanded.naming.column(property)
+      (expanded.ast match {
+        case ret: io.getquill.ast.ReturningAction =>
+            io.getquill.norm.ExpandReturning.applyMap(ret)(
+              (ast, statement) => io.getquill.context.Expand(${c.prefix}, ast, statement, idiom, naming).string
+            )(idiom, naming)
         case ast =>
```

Comment:
missing `case io.getquill.ast.Returning` ?

Reply:
So ReturningAction is a de-composer that matches Returning or ReturningGenerated, I defined it in `Ast.scala`. I decide to write it since their structure is identical. I'm not sure I like the name "ReturningAction" though. Maybe "ReturningAst"? Or maybe it shouldn't start with a capital letter. Any thoughts?